### PR TITLE
Runtime function

### PR DESCRIPTION
**Pull request by privat**
Refactor SeparateRuntimeFunction and related services in the separate compiler.

This simplify the compiler and improve the efficiency of the generated code a little.

Before: 0m7.588s
After: 0m7.364s (-3%)

This refactorisation is required for some next PR that will improve polymorphic calls

### CODE REVIEW

File: src/compiler/separate_compiler.nit [hunk 1]
```diff
@@ -1778,108 +1774,112 @@ class SeparateCompilerVisitor
 end
 
 redef class MMethodDef
-	fun separate_runtime_function: AbstractRuntimeFunction
+	# The C function associated to a methoddef
```
Comment (Morriar) [inline src/compiler/separate_compiler.nit]: MMethodDef ou AMethPropdef?

File: src/compiler/separate_compiler.nit [hunk 2]
```diff
@@ -1778,108 +1774,112 @@ class SeparateCompilerVisitor
...
+			res.is_thunk = true
 		end
 		return res
 	end
-	private var virtual_runtime_function_cache: nullable VirtualRuntimeFunction
+	private var virtual_runtime_function_cache: nullable SeparateRuntimeFunction
+end
+
+redef class MSignature
+	# Does the C-version of `self` the same than the C-version of `other`
```
Comment (Morriar) [inline src/compiler/separate_compiler.nit]: Missing the final `?`

**Comment by Morriar** [comment]
+1

**Comment by privat** [comment]
reroll: remarques d'alex + ajout d'un commit pour réparer --erasure

**Comment by R4PaSs** [comment]
+1

**Pull request merged**